onnxmltools/convert/coreml/shape_calculators/Regressor.py

```diff
@@ -34,8 +35,10 @@ def calculate_traditional_regressor_output_shapes(operator):
         raise ValueError('Model should be one of linear model, tree-based model, and support vector machine')
 
     N = operator.inputs[0].type.shape[0]
-    operator.outputs[0].type = FloatTensorType([N, C], doc_string=operator.outputs[0].type.doc_string)
-
+    if operator.targeted_onnx_version < StrictVersion('1.2') or C > 1:
+        operator.outputs[0].type = FloatTensorType([N, C], doc_string=operator.outputs[0].type.doc_string)
+    else:
+        operator.outputs[0].type = FloatTensorType([N], doc_string=operator.outputs[0].type.doc_string)
```

Comment:
There are some more files which may need to be changed. For example,  scikit-learn [linear classifier](https://github.com/onnx/onnxmltools/blob/master/onnxmltools/convert/sklearn/shape_calculators/LinearClassifier.py), [SVM](https://github.com/onnx/onnxmltools/blob/master/onnxmltools/convert/sklearn/shape_calculators/SVM.py).

Reply:
ok, but I am a little doubting that whether onnx can support [N, C] output where C > 1.

Outputs
Y : tensor(float)
N classes

Reply:
It's not documented, so anything can happen. Maybe double-check with @linkerzhang and clarify in ONNX spec. Please make sure the final definition you reach doesn't not violate the released runtime if possible.